### PR TITLE
Update mozjs to 87

### PR DESCRIPTION
**Pull request by sagudev**
Depends on: https://github.com/servo/rust-mozjs/pull/540

### CODE REVIEW

File: components/script/dom/imagedata.rs [hunk 1]
```diff
@@ -113,7 +114,12 @@ impl ImageData {
         let len = width * height * 4;
         let cx = global.get_cx();
         rooted!(in (*cx) let mut array = ptr::null_mut::<JSObject>());
-        Uint8ClampedArray::create(*cx, CreateWith::Length(len), array.handle_mut()).unwrap();
+        Uint8ClampedArray::create(
+            *cx,
+            CreateWith::Length(len.try_into().unwrap()),
```
Comment (jdm) [inline components/script/dom/imagedata.rs:119]: `len as usize` doesn't require any unwrap.

**Comment by sagudev** [comment]
Ready for merge!

**Comment by CYBAI** [comment]
This needs a format 🎨 

```
Run `./mach fmt` to fix the formatting
```

**Pull request merged**